### PR TITLE
Add Raku lexer

### LINKED ISSUE #306: Add support for Raku programming language
Raku(formerly known as Perl 6) - https://raku.org/

### PR DESCRIPTION
**Pull request by CIAvash**
Fixes #306

### CODE REVIEW

**Comment by alecthomas** [comment]
Thanks, but CI failed.

**Comment by alecthomas** [comment]
Thanks, but CI is failing.

**Comment by alecthomas** [comment]
Also can you add a test? See lexers/README.md for details.

**Comment by CIAvash** [comment]
Circleci needs logging in, I find that ridiculous, so I can't see what is failing. When I used the Chroma command line tool there was no error.

**Comment by alecthomas** [comment]
Yes, that is annoying I agree. The failure is linter failures. You'll need golangci-lint 1.26.0 to check this locally.

```
lexers/r/raku.go:1447: unnecessary leading newline (whitespace)
lexers/r/raku.go:1142:5: ifElseChain: rewrite if-else to switch statement (gocritic)
lexers/r/raku.go:1414:2: ifElseChain: rewrite if-else to switch statement (gocritic)
lexers/r/raku.go:1232:15: Error return value of `podFormatter` is not checked (errcheck)
lexers/r/raku.go:756:73: `archtypes` is a misspelling of `archetypes` (misspell)
lexers/r/raku.go:639:2: exported const RakuQuote should have comment (or a comment on this block) or be unexported (golint)
lexers/r/raku.go:1146:6: should replace `nestingLevel += 1` with `nestingLevel++` (golint)
lexers/r/raku.go:1150:6: should replace `nestingLevel -= 1` with `nestingLevel--` (golint)
lexers/r/raku.go:1252:2: exported const TopRule should have comment (or a comment on this block) or be unexported (golint)
lexers/r/raku.go:17: File is not `gofmt`-ed with `-s` (gofmt)
```

**Comment by CIAvash** [comment]
This is unrelated, it says change `ifElseChain ` to `switch`, but then what is the point of `if else`?

File: README.md [hunk 1]
```diff
@@ -260,7 +260,7 @@ go get -u github.com/alecthomas/chroma/cmd/chroma
 
 - Quite a few lexers, for various reasons (pull-requests welcome):
     - Pygments lexers for complex languages often include custom code to
-      handle certain aspects, such as Perl6's ability to nest code inside
```
Comment (alecthomas) [inline README.md:263]: This sentence now no longer makes sense?

**Review by alecthomas — APPROVED** [review]
Wow, I just looked at the actual code. I hadn't realised that you'd manually written this lexer. Impressive work, thank you!

Just one small comment, but otherwise LGTM.

File: README.md (hunk 1 above)
Comment (CIAvash) [reply README.md:263]: I only removed the perl6 part and missed the regular expressions part.
Do you mean the whole "Pygments lexers for complex languages often include custom code to..." part. or just that "regular expressions part"?

**Comment by CIAvash** [comment]
> Wow, I just looked at the actual code. I hadn't realised that you'd manually written this lexer. Impressive work, thank you!
> 
> Just one small comment, but otherwise LGTM.

Hehe, yeah it took me about two to three weeks! I started from the lexer generated by `pygments2chroma.py`, but changed and added a lot of things

File: README.md (hunk 1 above)
Comment (alecthomas) [reply README.md:263]: I just mean it now reads "handle certain aspects regular expressions", which doesn't make sense. Maybe just revert this, or did you mean to change it to Raku?

**Comment by alecthomas** [comment]
Awesome, thanks!

**Comment by alecthomas** [comment]
Sorry, I meant to merge this!

**Pull request merged**